erpnext/projects/doctype/timesheet/timesheet.py

```diff
@@ -28,6 +28,7 @@ def validate(self):
 		self.update_cost()
 		self.calculate_total_amounts()
 		self.calculate_percentage_billed()
+		self.set_dates()
```

Comment:
Adding function under validate will execute on every save. Start date and end date is used in salary slip while making it from timesheet